frontends/p4/typeChecking/typeChecker.cpp

```diff
@@ -2836,6 +2892,10 @@ void TypeInference::checkCorelibMethods(const ExternMethod* em) const {
 const IR::Node* TypeInference::postorder(IR::MethodCallExpression* expression) {
     if (done()) return expression;
     LOG2("Solving method call " << dbp(expression));
+    auto method = expression->method;
```

Comment:
This is clearly wrong. The type of a method call is not the method type, but the result of the method type.

Reply:
In the Member postorder method where sz is computed, the result type is set to Type_Bits.  Subsequently the MethodCallExpression is run.  I think it is semantics because once the type was set, it is checked for the same type here and then method is returned.  What would be a code change you would suggest?